main/config.c

```diff
@@ -258,14 +258,20 @@ struct ast_config {
 	struct ast_config_include *includes;  /*!< a list of inclusions, which should describe the entire tree */
 };
 
+struct ast_config_included_file {
+	AST_LIST_ENTRY(ast_config_included_file) next;
+	/*|< #include path or matching #tryinclude paths */
+	char fn[0];
```

Comment:
Don't include 0, as that can cause gcc to complain, just `char fn[];` to declare a flexible struct member
You also want const char *filename, and just leave the last member named as "data" per convention.

Reply:
I was following other examples in the same code (struct ast_comment, struct cache_file_include, cache_file_mtime).

I can switch from "char fn[0]" to "char fn[]" but I can't use "const char fn[]" and continue to use strcpy().  Or, I can leave the code as-is and match the other usage in config.c

Note: if you are thinking about doing this a different way then can you point me to an example.

Reply:
> I was following other examples in the same code (struct ast_comment, struct cache_file_include, cache_file_mtime).
> 
> I can switch from "char fn[0]" to "char fn[]" but I can't use "const char fn[]" and continue to use strcpy(). Or, I can leave the code as-is and match the other usage in config.c
> 
> Note: if you are thinking about doing this a different way then can you point me to an example.

Here's an example where it was done twice (a bit more complicated since it involves pointer math), but in this case it'd just be once: https://github.com/InterLinked1/phreakscript/blob/309fc67c71589c04b43f69b775391a52085fa1c4/res/res_alarmsystem.c#L2030

You have both a const char * and a char[], the const char * is set to the char after the strcpy.

Interesting that you see char[0] in the code, that's technically not correct and I'm surprised it's not triggering compiler warnings with some versions of gcc. I've had that happen in the past where I had to change [0] to []

Reply:
Yeah, you don't need the const char pointer, strictly, the advantage is mainly semantic since application code would be using the const char rather than the char directly. If you want to just use the FSM directly I suppose that's fine given the limited use of the struct.